### ISSUE TITLE
Error: Cannot find module 'graphile-build-pg'

### ISSUE THREAD

**Issue by tyvdh** [graphile/postgraphile#778]
I'm submitting a ... 

* [x] bug report
* [ ] feature request
* [ ] question 

PostGraphile version: `4.0.0-rc.1` or whatever `npm install -g postgraphile` is handing out these days.

Running command via a Docker file. Has been working flawlessly for months. Something about the update has it messed up.
```bash
postgraphile -n 0.0.0.0 -c postgres://$USER:$PASS@$HOST:$PORT/$DB -o -a -j -M --append-plugins /plugins/connection-filter/index.js
```

Getting this output
```bash
Error: Cannot find module 'graphile-build-pg'
at Function.Module._resolveFilename (internal/modules/cjs/loader.js:594:15)
at Function.Module._load (internal/modules/cjs/loader.js:520:25)
at Module.require (internal/modules/cjs/loader.js:650:17)
at require (internal/modules/cjs/helpers.js:20:18)
at Object.<anonymous> (/plugins/connection-filter/src/PgConnectionArgFilterPlugin.js:1:80)
at Module._compile (internal/modules/cjs/loader.js:702:30)
at Object.Module._extensions..js (internal/modules/cjs/loader.js:713:10)
at Module.load (internal/modules/cjs/loader.js:612:32)
at tryModuleLoad (internal/modules/cjs/loader.js:551:12)
at Function.Module._load (internal/modules/cjs/loader.js:543:3)
```

Happy to provide any further helpful information. Seems like it could likely be an easy fix.

**Comment by tyvdh** [comment]
Solved. Sort of.

I had to manually add `graphile-build-pg`, `graphql` and `pg` to my Dockerfile.

```bash
npm install -g postgraphile && \
npm install graphile-build-pg && \
npm install graphql && \
npm install pg && \
```

Seems odd those aren't included with the `postgraphile` package, but maybe this is intentional?

**Comment by benjie** [comment]
They’re included via postgraphile-core; it should work... are you able to share your dockerfile? The official graphile/postgraphile docker image was working last I checked.

**Comment by tyvdh** [comment]
Yeah it’s OS. https://github.com/TinyAnvil/postgraphile-docker

**Comment by tyvdh** [comment]
Thanks @benjie! Works like a charm now.

**Comment by garretteklof** [comment]
Hi Benjie - I'm going to leave this here as opposed to opening a new issue. I'm currently running into something similar. I'm unable to isolate where it's coming from (one of the plugins or `postgraphile-apollo-server`, which I am also using). If I explicitly add the `graphile-build-pg` dependency the server works as expected. I blindly updated `postgraphile` and related plugins, but when I reverted back the error was still there. I think this has to do with me deleting my old node_modules in the process. Somewhere along the line the dependency must have snuck in there? Still not sure why it's explicitly needed if it's included via core. Any insight on this would be greatly appreciated! Thanks.

`Error: Cannot find module 'graphile-build-pg'
Require stack:
- /Users/garretteklof/_dev/agrograph/agros-api/node_modules/graphile-utils/node8plus/makePgSmartTagsPlugin.js
- /Users/garretteklof/_dev/agrograph/agros-api/node_modules/graphile-utils/node8plus/index.js
- /Users/garretteklof/_dev/agrograph/agros-api/node_modules/postgraphile/build/index.js
- /Users/garretteklof/_dev/agrograph/agros-api/node_modules/postgraphile/index.js
- /Users/garretteklof/_dev/agrograph/agros-api/node_modules/postgraphile-apollo-server/index.js
- /Users/garretteklof/_dev/agrograph/agros-api/server/index.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:980:15)
    at Function.Module._load (internal/modules/cjs/loader.js:862:27)
    at Module.require (internal/modules/cjs/loader.js:1040:19)
    at require (internal/modules/cjs/helpers.js:72:18)
    at Object.<anonymous> (/Users/garretteklof/_dev/agrograph/agros-api/node_modules/graphile-utils/src/makePgSmartTagsPlugin.ts:2:1)
    at Module._compile (internal/modules/cjs/loader.js:1151:30)
    at Module._compile (/Users/garretteklof/_dev/agrograph/agros-api/node_modules/pirates/lib/index.js:99:24)
    at Module._extensions..js (internal/modules/cjs/loader.js:1171:10)
    at Object.newLoader [as .js] (/Users/garretteklof/_dev/agrograph/agros-api/node_modules/pirates/lib/index.js:104:7)
    at Module.load (internal/modules/cjs/loader.js:1000:32) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/garretteklof/_dev/agrograph/agros-api/node_modules/graphile-utils/node8plus/makePgSmartTagsPlugin.js',
    '/Users/garretteklof/_dev/agrograph/agros-api/node_modules/graphile-utils/node8plus/index.js',
    '/Users/garretteklof/_dev/agrograph/agros-api/node_modules/postgraphile/build/index.js',
    '/Users/garretteklof/_dev/agrograph/agros-api/node_modules/postgraphile/index.js',
    '/Users/garretteklof/_dev/agrograph/agros-api/node_modules/postgraphile-apollo-server/index.js',
    '/Users/garretteklof/_dev/agrograph/agros-api/server/index.js'
  ]
}
[nodemon] app crashed - waiting for file changes before starting...`

**Comment by benjie** [comment]
@garretteklof I suggest you use yarn to install your dependencies, and then run `npx yarn-deduplicate && yarn`. That tends to fix issues like this.

**Comment by garretteklof** [comment]
Cool - `yarn deduplicate` worked. Thanks a lot!
